From dc99194a6ac9f7fab909da23606adcc45e45b78f Mon Sep 17 00:00:00 2001
From: Kate Hsuan <hpa@redhat.com>
Date: Wed, 24 Jan 2024 15:40:43 +0800
Subject: [PATCH 1/2] src: thd_dbus_interface: Fix unable to terminate through
 Dbus API
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Setup an appropriate exit_handler to make sure thermald can be properly
terminated.
---
 src/thd_dbus_interface.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/thd_dbus_interface.cpp b/src/thd_dbus_interface.cpp
index 18c0e81..67f0044 100644
--- a/src/thd_dbus_interface.cpp
+++ b/src/thd_dbus_interface.cpp
@@ -1220,6 +1220,8 @@ int thd_dbus_server_init(gboolean (*exit_handler)(void)) {
 		return THD_FATAL_ERROR;
 	}
 
+	thd_dbus_exit_callback = exit_handler;
+
 	interface_vtable.method_call = thd_dbus_handle_method_call;
 	interface_vtable.get_property = thd_dbus_handle_get_property;
 	interface_vtable.set_property = thd_dbus_handle_set_property;
-- 
2.39.2

